rpc/migration/oidc_to_email.go

```diff
@@ -0,0 +1,176 @@
+package migration
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/0xsequence/waas-authenticator/config"
+	"github.com/0xsequence/waas-authenticator/data"
+	"github.com/0xsequence/waas-authenticator/proto"
+	"github.com/0xsequence/waas-authenticator/rpc/attestation"
+	"github.com/0xsequence/waas-authenticator/rpc/auth/email"
+	"github.com/0xsequence/waas-authenticator/rpc/crypto"
+	"github.com/0xsequence/waas-authenticator/rpc/tenant"
+)
+
+type OIDCToEmail struct {
+	accounts *data.AccountTable
+	tenants  *data.TenantTable
+	config   config.EmailMigrationConfig
+}
+
+func (m *OIDCToEmail) OnRegisterSession(ctx context.Context, originalAccount *data.Account) error {
```

Comment:
when is `OnRegisterSession` called for the migration?

Reply:
It's called in the `RegisterSession` method, at the very end: https://github.com/0xsequence/waas-authenticator/blob/b22b70e9daf95f2c2a15fbf2aa5ad7b3ebb31fb3/rpc/sessions.go#L204

The migrations runner decides which migrations are relevant and executes those.

Reply:
do we really need this..? because, we will run this once and never again for this migration..?